sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -312,6 +312,7 @@ case class AdaptiveSparkPlanExec(
   }
 
   override def doExecuteBroadcast[T](): broadcast.Broadcast[T] = {
+    assert(getFinalPhysicalPlan().isInstanceOf[BroadcastExchangeExec])
     getFinalPhysicalPlan().doExecuteBroadcast()
```

Comment:
`getFinalPhysicalPlan()` seems heavy. Can we change it to the following:
```scala
val plan = getFinalPhysicalPlan()
assert(plan.isInstanceOf[BroadcastExchangeExec])
plan.doExecuteBroadcast()
```

Reply:
Updated.